### ISSUE TITLE
[libc][docs] Should we add types, symbolic constants to implementation status docgen pages?

### ISSUE THREAD

**Issue by Flandini** [llvm/llvm-project#89667]
Addressing comments from https://github.com/llvm/llvm-project/pull/89421#issuecomment-2070949464.

Tagging @nickdesaulniers

**Comment by nickdesaulniers** [comment]
Probably.  Do you have examples of some of these?  The one concern I have is if any one header defines some ridiculous number of either, it may look verbose to document all of them. But perhaps that concern is unwarranted.

**Comment by Flandini** [comment]
Yeah some have a lot of symbolic constants. Just looking at `signal.h` in POSIX2018, these would be added:

### Types:

sig_atomic_t
sigset_t
pid_t
sigevent
sigval
sigaction
mcontext_t
ucontext_t
stack_t
siginfo_t

### Symbolic Constants
SIGEV_NONE
SIGEV_SIGNAL
SIGEV_THREAD
SA_NOCLDSTOP
SA_ONSTACK
SA_RESETHAND
SA_RESTART
SA_SIGINFO
SA_NOCLDWAIT
SA_NODEFER
SS_ONSTACK
SS_DISABLE
MINSIGSTKSZ
SIGSTKSZ
ILL_ILLOPC
ILL_ILLOPN
ILL_ILLADR
ILL_ILLTRP
ILL_PRVOPC
ILL_PRVREG
ILL_COPROC
ILL_BADSTK
FPE_INTDIV
FPE_INTOVF
FPE_FLTDIV
FPE_FLTOVF
FPE_FLTUND
FPE_FLTRES
FPE_FLTINV
FPE_FLTSUB
SEGV_MAPERR
SEGV_ACCERR
BUS_ADRALN
BUS_ADRERR
BUS_OBJERR
TRAP_BRKPT
TRAP_TRACE
CLD_EXITED
CLD_KILLED
CLD_DUMPED
CLD_TRAPPED
CLD_STOPPED
CLD_CONTINUED
POLL_IN
POLL_OUT
POLL_MSG
POLL_ERR
POLL_PRI
POLL_HUP
SI_USER
SI_QUEUE
SI_TIMER
SI_ASYNCIO
SI_MESGQ

**Comment by Flandini** [comment]
Looking through more of POSIX2018:
[`unistd.h`](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/unistd.h.html) probably has the most, then [`sys/types.h`](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html), [`limits.h`](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html). 

A few other headers have a bit less than those, and the rest don't have many other type or constant definitions.
